src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/BlockEditorValueConnector.cs

```diff
@@ -0,0 +1,245 @@
+                    // throws if not found - no need for a null check
+                    var propValueConnector = ValueConnectors.Get(propType);
+
+                    // pass the value, property type and the dependencies collection to the connector to get a "artifact" value
+                    var val = block.PropertyValues[key];
+                    object parsedValue = propValueConnector.ToArtifact(val, propType, dependencies);
+
+                    // getting Map image value umb://media/43e7401fb3cd48ceaa421df511ec703c to (nothing) - why?!
+                    _logger.Debug<BlockEditorValueConnector>("Map " + key + " value '" + block.PropertyValues[key] + "' to '" + parsedValue + "' using " + propValueConnector.GetType() + " for " + propType);
+
+                    parsedValue = parsedValue?.ToString();
+
+                    block.PropertyValues[key] = parsedValue;
+                }
+            }
+
+            value = JsonConvert.SerializeObject(blockEditorValue);
+            return (string)value;
+        }
+
+        public object FromArtifact(string value, PropertyType propertyType, object currentValue)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return value;
+            }
+
+            if (value.DetectIsJson() == false)
+                return value;
+
+            var blockEditorValue = JsonConvert.DeserializeObject<BlockEditorValue>(value);
+
+            if (blockEditorValue == null)
+                return value;
+
+            var allContentTypes = blockEditorValue.Data.Select(x => x.ContentTypeKey)
+                .Distinct()
+                .ToDictionary(a => a, a =>
+                {
+                    if (!Guid.TryParse(a, out var keyAsGuid))
+                        throw new InvalidOperationException($"Could not parse ContentTypeKey as GUID {keyAsGuid}.");
+                    return _contentTypeService.Get(keyAsGuid);
+                });
+
+            //Ensure all of these content types are found
+            if (allContentTypes.Values.Any(contentType => contentType == null))
+            {
+                throw new InvalidOperationException($"Could not resolve these content types for the Block Editor property: {string.Join(",", allContentTypes.Where(x => x.Value == null).Select(x => x.Key))}");
+            }
+
+            foreach (var block in blockEditorValue.Data)
+            {
+                var contentType = allContentTypes[block.ContentTypeKey];
+
+                foreach (var key in block.PropertyValues.Keys.ToArray())
+                {
+                    var innerPropertyType = contentType.CompositionPropertyTypes.FirstOrDefault(x => x.Alias == key);
+
+                    if (innerPropertyType == null)
+                    {
+                        _logger.Debug<BlockEditorValueConnector>($"No property type found with alias {key} on content type {contentType.Alias}");
```

Comment:
```suggestion
                        _logger.Debug<BlockEditorValueConnector>("No property type found with alias {Key} on content type {ContentTypeAlias}", key, contentType.Alias);
```

Use the nice serilogger we have in V8 please & pass the params so they can be easily found in logs